### PR TITLE
Update turtlebot2_demo.repos to match latest ros2.repos.

### PR DESCRIPTION
**Pull request by clalancette**
This should actually make it compile again, hopefully.

Signed-off-by: Chris Lalancette <clalancette@osrfoundation.org>

connects to ros2/ros2#342

### CODE REVIEW

**Review by wjwwood — APPROVED** [review]

File: turtlebot2_demo.repos
Comment (mikaelarguedas) [inline turtlebot2_demo.repos:45]: if we use fastRTPS master here I think it would make sense to move the boost dependencies installation in the [turtlebot section of the dockerfile](https://github.com/ros2/ci/blob/master/linux_docker_resources/Dockerfile#L89). If they are needed for the turtlebot2_demo, otherwise they can be removed.

**Comment by gerkey** [comment]
I've been using this version of the `turtlebot2_demo.repos` file to test the CI job (#16, #17, and ros2/ci#63), which is now working. So, +1 to merging this after others' comments are addressed.

**Comment by gerkey** [comment]
@clalancette can I help to wrap up this PR?

**Comment by clalancette** [comment]
@gerkey So, I think the only outstanding comment is to move the boost dependencies into the turtlebot-specific portion of the ci scripts.  I'm going to merge this PR, then open one against ros2/ci to do that.

**Comment by gerkey** [comment]
Great, thanks. Here's a build using the job's default args to check that CI will now work as expected:

http://ci.ros2.org/job/ci_turtlebot-demo/22/

**Comment by clalancette** [comment]
And here's a PR (not yet ready for review) to move boost into the turtlebot section only: ros2/ci#67

**Comment by gerkey** [comment]
The post-merge build looks good. Just one compiler warning, which will disappear after ros2/ros_astra_camera#4 is merged (waiting on approval for that).

**Comment by clalancette** [comment]
Will take a look at that one right now.

**Pull request merged**
